### PR TITLE
feat(terminal): smart Ctrl+C/Ctrl+V copy/paste

### LINKED ISSUE #607: [Enhancement]: Smart Ctrl+C/V for copy/paste without Shift (like Windows Terminal)
**Is your feature request related to a problem?**
In WSL sessions, native commands like copy and paste require appending the `Shift` key (e.g., `Ctrl+Shift+C` / `Ctrl+Shift+V`). This is an extra step compared to Windows Terminal, which handles this more seamlessly.

**Describe the solution you'd like:**
Implement "smart" copy/paste shortcuts. If text is highlighted in the `xterm.js` buffer, pressing `Ctrl+C` should copy the text. If no text is selected, `Ctrl+C` should send the standard `SIGINT` (`^C`) to the PTY. `Ctrl+V` should paste directly. This matches the modern developer experience expected from tools like Windows Terminal, removing the friction of remembering to add `Shift`.

### PR DESCRIPTION
**Pull request by Gimics**
## What
Adds Windows-Terminal-style clipboard behavior in the terminal on Windows and Linux: Ctrl+C copies the selection when text is selected and otherwise sends SIGINT, and Ctrl+V pastes. Ctrl+Shift+C / Ctrl+Shift+V keep copying/pasting unconditionally. Gated behind a new `terminalSmartCopyPaste` setting (default on).

## Why
Closes #607. On Windows/Linux, Ctrl+C only sent SIGINT and Ctrl+V inserted a literal control char, so the clipboard required Ctrl+Shift+C/V. This matches what people expect from Windows Terminal while keeping Ctrl+C as a reliable interrupt when nothing is selected.

## How
- New pure function `terminalClipboardIntent(event, opts)` in `terminal/lib/keymap.ts` returns `"copy" | "paste" | null` from the key event, the preference, and whether there is a selection. Unit-tested.
- `terminal/lib/rendererPool.ts` calls it inside the existing `attachCustomKeyEventHandler` and acts on the result; the old `isTerminalCopy`/`isTerminalPaste` helpers are removed. A smart (no-Shift) copy clears the selection so a second Ctrl+C interrupts; Ctrl+Shift+C keeps the selection.
- Preference `terminalSmartCopyPaste` is plumbed through `settings/store.ts` and pushed to the pool via `applySmartCopyPaste` in `useTerminalSession.ts`, mirroring the existing cursor-blink/webgl wiring.
- Settings toggle under General > Terminal (hidden on macOS).
- No new dependencies and no IPC: paste reuses the same `navigator.clipboard` path Ctrl+Shift+V already used.

macOS is intentionally untouched: Cmd+C/Cmd+V stay native and Ctrl+C stays SIGINT, so the logic early-returns there.

## Testing
Added `terminalClipboardIntent` unit tests (macOS no-op, Ctrl+Shift+C/V, smart Ctrl+C with/without selection, smart Ctrl+V, smart-off, AltGr/other-key, uppercase key form). Manually exercised in `pnpm tauri dev` on Windows/pwsh: selection + Ctrl+C copies and clears the highlight; Ctrl+C with no selection interrupts a running command; Ctrl+V pastes; Ctrl+Shift+C/V still work; toggling the setting off restores Ctrl+C-always-interrupts live.

- [x] `pnpm exec tsc --noEmit` clean
- [x] Manual smoke-test of the affected feature
- [x] (If you touched `src-tauri/`) `cargo test`/`clippy` -- N/A, no Rust changes
- [x] (If you changed a `#[tauri::command]` signature) -- N/A
- [x] (If UI) tested in `pnpm tauri dev`
- [x] Platforms tested: Windows (Linux shares the same non-mac path; macOS is a no-op by design)
- [x] Shells tested: pwsh / powershell

## Screenshots / GIFs
The only new on-screen UI is the settings toggle (General > Terminal):

image

The rest of the change is keyboard behavior (Ctrl+C / Ctrl+V / Ctrl+Shift+C / Ctrl+Shift+V), which a recording cannot show meaningfully without a keystroke overlay; the exact flows are in Testing above.

## Notes for reviewer
- Default is **on** to match Windows Terminal and #607. This changes default behavior on Windows/Linux, but only in the narrow case of pressing Ctrl+C while text is selected. Happy to flip to opt-in (default off) if you would prefer no behavior change on update.
- Kept paste on `navigator.clipboard` (no `tauri-plugin-clipboard-manager`) to avoid a new dependency and per-paste IPC, consistent with the existing Ctrl+Shift+V code and the OSC 52 handler. On WebView2 the first `readText()` shows a one-time clipboard-read permission prompt; that is pre-existing (Ctrl+Shift+V hit it too) and out of scope here.
- The empty `.catch(() => {})` on clipboard read/write matches the existing pattern in this file and `osc-handlers.ts`; left as-is for consistency.

## Summary by CodeRabbit

* **New Features**
  * Added a new **Smart copy / paste** terminal setting in **General** that controls clipboard behavior for Ctrl-based copy/paste shortcuts.
  * When enabled, Ctrl+V always triggers paste, and Ctrl+C triggers copy only when text is selected; Shift+Ctrl/Cmd versions always perform copy/paste.
  * The setting is not available on macOS.

* **Tests**
  * Expanded clipboard-intent detection tests to cover smart vs non-smart modes, selection gating, macOS-specific behavior, non-Latin layouts, and additional key-combination edge cases.

### CODE REVIEW

File: src/modules/terminal/lib/keymap.ts [hunk 1]
```diff
@@ -43,3 +43,43 @@ export function terminalDeleteSequence(
...
+ * null there and never shadows the interrupt.
+ */
+export function terminalClipboardIntent(
+  event: TerminalKeyEvent,
+  opts: ClipboardOpts,
+): TerminalClipboardIntent | null {
+  if (opts.isMac) return null;
+  if (!event.ctrlKey || event.altKey || event.metaKey) return null;
+  const isC = event.code === "KeyC" || event.key === "c" || event.key === "C";
+  const isV = event.code === "KeyV" || event.key === "v" || event.key === "V";
```
[the remark this answers is not in the document]
Comment (Gimics) [reply src/modules/terminal/lib/keymap.ts]: The flag here is valid for the Dvorak case: `event.code` keys off physical position, so on a Dvorak-style layout `Ctrl+<physical-C>` can trigger copy. I kept the `event.code` check deliberately, though: dropping it and matching only `event.key` would stop Ctrl+C/Ctrl+V from working on non-Latin layouts (Cyrillic, Greek, etc.), where the physical C/V keys produce non-Latin characters and `event.key` is never `"c"`/`"v"`. Matching physical position is what keeps the shortcuts working there (same reason editors like VS Code key these off physical position). AltGr combos are already excluded earlier by the `event.altKey` guard, so the `ç`-style case never reaches this code.

This is also the pre-existing behavior of the copy/paste handler this PR refactors, so I'd rather not regress it here.

I did take the tidiness half of the suggestion: collapsed the uppercase checks into `event.key.toLowerCase()`, and added a test locking the non-Latin behavior (Cyrillic `с`/`м` with `code` `KeyC`/`KeyV` still copy/paste) so the intent is explicit.

**Comment by Gimics** [comment]
For triage context, since terminal clipboard is a busy area right now, here is how this PR is scoped relative to the other open work:

 - Focus: this only changes which keystrokes trigger copy/paste — a pure terminalClipboardIntent decision function plus the existing attachCustomKeyEventHandler wiring. No new dependencies, no IPC; paste reuses the same navigator.clipboard path Ctrl+Shift+V already used.
 - Overlaps #776: that PR also adds selection-aware Ctrl+C copy, but bundles image paste and a Thai IME fix and has been idle since June 11. This is the focused, keyboard-only slice of that behavior, with unit tests. Happy to defer if Piya-Boy is still active on it.
 - Orthogonal to #724 / #712 (native clipboard manager): those change how the clipboard is read/written; this changes when copy and paste fire. They should compose rather than conflict.
 - Not the same as: image paste (#192, #685, #248), right-click copy/paste (#563), or copy-text formatting (#578).

If you would rather fold all of this into one holistic clipboard pass, or take a different direction for #607, say the word and I will close or rework this. Otherwise, it is ready for review.
